test/callback.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { SgidClient } from '../src'
+
+import {
+  MOCK_ACCESS_TOKEN,
+  MOCK_AUTH_CODE,
+  MOCK_CLIENT_ID,
+  MOCK_CLIENT_PRIVATE_KEY,
+  MOCK_CLIENT_SECRET,
+  MOCK_CODE_VERIFIER,
+  MOCK_HOSTNAME,
+  MOCK_REDIRECT_URI,
+  MOCK_SUB,
+} from './mocks/constants'
+import { tokenHandlerNoSubV2, tokenHandlerNoTokenV2 } from './mocks/handlers'
+import { server } from './mocks/server'
+
+describe('SgidClient: callback (with PKCE)', () => {
+  let client: SgidClient
+
+  beforeAll(() => {
+    client = new SgidClient({
+      clientId: MOCK_CLIENT_ID,
+      clientSecret: MOCK_CLIENT_SECRET,
+      privateKey: MOCK_CLIENT_PRIVATE_KEY,
+      redirectUris: [MOCK_REDIRECT_URI],
+      hostname: MOCK_HOSTNAME,
+    })
+  })
+
+  it('should call token endpoint and return sub and accessToken', async () => {
+    const { sub, accessToken } = await client.callback({
+      code: MOCK_AUTH_CODE,
+      codeVerifier: MOCK_CODE_VERIFIER,
+    })
+
+    expect(sub).toBe(MOCK_SUB)
+    expect(accessToken).toBe(MOCK_ACCESS_TOKEN)
+  })
+
+  it('should throw when no code verifier is provided', async () => {
+    await expect(client.callback({ code: MOCK_AUTH_CODE })).rejects.toThrow(
+      "Code verifier must be provided in 'callback' when using apiVersion 2",
```

Comment:
See comments in #35 on api version

Also, ultimately both PRs should be rebased against #27 / #27 should be merged into both PRs so that all error messages are in a central file. We can set up a session to go through the rebase together

Reply:
yup i definitely agree on the rebase with #27